src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -9,14 +9,18 @@ import { isUndefinedOrNull } from 'vs/base/common/types';
 import { assign } from 'vs/base/common/objects';
 import { find } from 'vs/base/common/arrays';
 
-export interface IConnectionProfileGroup {
-	id: string;
+export interface INewConnectionProfileGroup {
+	id?: string;
```

Comment:
Hmmmmm I know our code is written in a way that's making this necessary but can we fix it so ID isn't optional? :( (Maybe as part of a different PR)

Reply:
When creating a new profile group, the id isn't optional since it gets generated when saved to the settings. Thats why I split the create new profile group to take a different interface where the id is optional instead of just passing undefined when the id is specified as required.